File: bblaser-service/src/main/java/nl/sijpesteijn/bblaserservice/repositores/AnimationRepository.java
package nl.sijpesteijn.bblaserservice.repositores;

import nl.sijpesteijn.bblaserservice.domain.BBAnimation;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface AnimationRepository extends MongoRepository<BBAnimation, String> {
}

File: bblaser-service/src/main/kotlin/nl/sijpesteijn/bblaser/repositories/AnimationRepository.kt
package nl.sijpesteijn.bblaser.repositories

import nl.sijpesteijn.bblaser.BBAnimation
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.mongodb.repository.Query
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository


@Repository
interface AnimationRepository : PagingAndSortingRepository<BBAnimation, String> {

    @Query("{ _id: { \$exists: true }}")
    fun retrieveAllAnimationsPaged(pageable: Pageable): Page<BBAnimation>
}